packages/components/src/custom-gradient-picker/index.js

```diff
@@ -53,7 +53,7 @@ const GradientAnglePicker = ( { gradientAST, hasGradient, onChange } ) => {
 		<AnglePickerControl
 			__nextHasNoMarginBottom
 			onChange={ onAngleChange }
-			labelPosition="top"
+			// What should we pass here instead of `''` ?
```

Comment:
As per comment — currently this usage of `AngleControlPicker` passes an empty string when the value is not defined, which is technically incorrect since `value` should have type `number`. 

A few alternative approaches:
 - Allow `value` to be of type `string | number`
 - Allow `value` (and therefore `onChange`) to be optional, and therefore pass `undefined` instead of `''`
 - Use `0` instead of `''`

@mirka , what do you think?

My instinct tells me that allowing `value` and `onChange` to be optional would probably be the wisest change here, but I'm definitely open to more thoughts on the matter.

Reply:
I think making `value`, and `onChange` optional aligns more closely with the previous usage here though I don't hold a strong opinion either.

Reply:
I agree keeping it optional makes most sense from a developer's perspective 👍

Reply:
So! I went ahead and marked `onChange` and `value` as optional: https://github.com/WordPress/gutenberg/pull/45820/commits/04511609dedb3eed00850af93a3dbbcd930478fa

This in turn revealed another issue: with `value` and `onChange` as required, the component could only be used in _controlled_ mode; marking those props as optional enables it to also be used in _uncontrolled_ mode. This exposes a React error as soon we interact with the component in the `GradientPicker`, as we switch from an uncontrolled to controlled input.

Normally, a the way to solve this issue is to pass `''` as the `value`, which initialises the component as controlled, while still not setting a value. But this is currently not compatible with `AnglePickerControl`'s signature.

This "contrast" between the `string` and `number` types for the `value` prop is not new — in fact, we've been discussing it for a while, and it mostly has to do with `NumberControl` and it related component. And the more I think about it, the more I get the feeling that the `string` type is necessary for `value`, if anything to enable these components not to have an actual value while being _controlled_.

While the long term solution is clearly to make a decision about the type of the `value` prop in `NumberControl` and apply it to its derivate components (including `AnglePickerControl`), I've been thinking about the best compromise for this PR:

- we could revert `value` and `onChange` to be required again, therefore avoid marking the component as usable in "uncontrolled" mode for now
- we could change value's type to to `number | string`, and therefore accept the `''` value that `CustomGradientPicker` is passing

At a later point, we could:

- make the component usable in uncontrolled mode (i.e. marking `value` and `onChange` as optional, and potentially use the [`useControlledValue` hook](https://github.com/WordPress/gutenberg/blob/5cca2e1a8021679aaf2fc713c1a2c1cce168c374/packages/components/src/utils/hooks/use-controlled-value.ts) internally and make any further adjustment if needed)
- adjust the type if needed once we make a decision on `NumberControl`'s type


Curious to hear y'all's thoughts!

Reply:
Thanks for the detailed summary of your findings so far. Life is rarely ever easy, is it 😁 

If there is a particular option that reduces the potential for rework, I'd probably lean in that direction for now. 

The same sort of logic would also suggest that thought and effort might be better directed to solving the problem at its source rather than agonising too much over which bandaid solution is best here.

Reply:
> Life is rarely ever easy, is it 😁

It doesn't look like, does it? 😅 

> thought and effort might be better directed to solving the problem at its source rather

I think you're right, and I also think that working on this PR gave us enough insight on the fact that we sort of have to use `string` for the type, if anything in order to support properly controlled and uncontrolled version of a component.

I've opened https://github.com/WordPress/gutenberg/issues/45949 to highlight a tentative plan of action and track progress on it. I will come back to this PR once I make the necessary pre-requisite changes